### PR TITLE
Enable GitHub Actions workflow to automate new release updates

### PR DESCRIPTION
**Pull request by wraschke**
This pull request's objective is to fully automate the vNext updates in ci.docker repos for an upcoming release.

Workflow run demonstrating objectives met: https://github.com/wraschke/ci.docker/actions/runs/6669536508/job/18127527479

Resulting pull request: https://github.com/wraschke/ci.docker/pull/20/files

Note that this is a run done only in my fork--that is the environment where I've done all my testing and development.

It's not possible to test this workflow file in origin until they are merged.

And note that this PR introduces only the workflow YML file into main. The script it invokes needs to live in vNext: see accompanying https://github.com/WASdev/ci.docker/pull/558.

### CODE REVIEW

File: .github/workflows/update_files_for_release.yml [hunk 1]
```diff
@@ -0,0 +1,62 @@
+# This workflow is triggered by manual inputs.
+
+name: Update files for the new release
+
+on:
+  workflow_dispatch:
+    inputs:
+      OLD_VERSION:
```
Comment (leochr) [inline .github/workflows/update_files_for_release.yml:8]: The release after a quarterly release (version ending in .3, .6, .9, .12) wouldn't have an "old" version. The field is marked as required. Just want to make sure that such scenarios would work fine?

**Review by leochr — COMMENTED** [review]
@wraschke Thanks for the PR. Looks fine to me. Just a comment/inquiry about post-quarterly releases.

File: .github/workflows/update_files_for_release.yml (hunk 1 above)
Comment (leochr) [reply .github/workflows/update_files_for_release.yml:8]: We won't always have "old" version. For example, after we release 23.0.0.12, we'll only add 24.0.0.1

**Comment by wraschke** [comment]
To make this comment accessible/viewable to Leo. I'm responding to his review question:

> The release after a quarterly release (version ending in .3, .6, .9, .12) wouldn't have an "old" version. The field is marked as required. Just want to make sure that such scenarios would work fine?

Hi, Leo. "Old version" is supposed to refer to the version that is currently released and present in the Docker files--the one that we're trying to upgrade away from.

For example, let's consider the timeline now: in October, we're preparing for 23.0.0.11; "old version" is 23.0.0.10.

Whenever I run the automation scripts, I always need to specify the "old version" information--otherwise, the scripts don't work.

Should I change the variable and description of workflow textfield to make the input more clear?

File: .github/workflows/update_files_for_release.yml (hunk 1 above)
Comment (leochr) [reply .github/workflows/update_files_for_release.yml:8]: As long as the folders for the quarterly releases are not removed before they are supposed to be, we are okay. It seems like that would be the case from your description.

**Review by leochr — APPROVED** [review]

**Review by leochr — APPROVED** [review]

**Comment by wraschke** [comment]
One important thing to note is that the "old" automation scripts never removed the no-longer supported, stale quarterly release directory. This was always done manually; I just confirmed with Chris Potter.

I will try to automate that work in another iteration of this feature.

**Pull request merged**